### PR TITLE
extensions: pull from github in OSS build

### PR DESCRIPTION
**Pull request by connor4312**
Fixes https://github.com/microsoft/vscode-internalbacklog/issues/2676

Pulls from Github releases unless the marketplace URL is set in the product.json.

### CODE REVIEW

**Comment by sandy081** [comment]
Assigning to @joaomoreno as he owns this code.

File: build/lib/extensions.ts [hunk 1]
```diff
@@ -232,6 +235,50 @@ export function fromMarketplace(extensionName: string, version: string, metadata
...
+
+	const packageJsonFilter = filter('package.json', { restore: true });
+
+	return remote([`/repos${new URL(repo).pathname}/releases/tags/v${version}`], {
+		base: 'https://api.github.com',
+		requestOptions: { headers: ghApiHeaders }
+	}).pipe(through2.obj(function (file, _enc, callback) {
+		const asset = JSON.parse(file.contents.toString()).assets.find((a: any) => a.name.endsWith('.vsix'));
+		if (!asset) {
+			callback(new Error(`Could not find vsix in release of ${repo} @ ${version}`));
```
Comment (joaomoreno) [inline build/lib/extensions.ts:265]: Missing return?

**Review by joaomoreno — APPROVED** [review]

**Review by raghavthind2005 — APPROVED** [review]

File: build/lib/extensions.ts (hunk 1 above)
Comment (joaomoreno) [inline build/lib/extensions.ts:265]:
```suggestion
			return callback(new Error(`Could not find vsix in release of ${repo} @ ${version}`));
```

Needs recompilation. 🙈

**Comment by randomir** [comment]
Resolves #141340.

**Pull request merged**